docs/library/jesd204/axi_jesd204_rx/index.rst

```diff
@@ -0,0 +1,1129 @@
+``S_AXI`` interface. The interface is synchronous to the ``s_axi_aclk``. The
+``s_axi_aresetn`` signal is used to reset the peripheral and should be asserted
+during system startup until the ``s_axi_aclk`` is active and stable.
+De-assertion of the reset signal should be synchronous to ``s_axi_aclk``.
+
+JESD204 Control Signals
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The ``sync`` and ``sysref`` signals correspond to the SYNC~ and SYSREF signals
+of the JESD204 specification.
+
+The ``sync`` signal is asserted by the peripheral during link initialization and
+must be connected to the corresponding JESD204 ADC converter devices on the same
+link.
+
+The ``sysref`` signal is generated externally and is optional. It is only
+required to achieve deterministic latency in subclass 1 mode operation. If the
+``sysref`` signal is not connected software needs to configure the peripheral
+accordingly to indicate this.
+
+When the ``sysref`` signal is used, in order to ensure correct operation, it is
+important that setup and hold of the external signal relative to the
+``device_clk`` signal are met. Otherwise, deterministic latency cannot be
+guaranteed.
+
+Transceiver Interface (RX_PHYn)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+For each lane, the peripheral has one corresponding RX_PHY interface. These
+interfaces accept the physical layer data from the downstream physical layer
+transceiver peripheral.
+
+The physical layer is responsible for clock recovery, character alignment,
+de-serialization as well an 8b10b decoding.
+
+.. _axi_jesd204_rx_user_data:
+ 
+User Data Interface (RX_DATA)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+User data is provided on the AXI4-Stream ``RX_DATA`` interface. The interface is
+a reduced AXI4-Stream interface and only features the TVALID flow control
+signal, but not the TREADY flow control signal. The behavior of the interface is
+as if the TREADY signal was always asserted. This means as soon as ``rx_valid``
+is asserted a continuous stream of user data must be accepted from ``rx_data``.
+
+.. wavedrom::
+   :align: center
+
+   {
+      signal: [
+         ['RX_DATA',
+            { name: "device_clk", wave: 'P.........' },
+            { name: "rx_data",  wave: "x...======", data: ["D0", "D1", "D2",
+            "D3", "D4", "..."] },
+            { name: 'rx_valid', wave: '0...1.....' },
+         ]
+      ],
+      foot: {
+         text:
+            ['tspan',{dx:'-45'}, 'Link Inicialization', ['tspan', {dx:'60'},
+            'User Data Phase'],],
+         }
+   }
+
+After reset and during link initialization the ``rx_valid`` signal is
+de-asserted. As soon as the User Data Phase is entered the ``rx_valid`` will be
+asserted to indicate that the peripheral is now providing and the processed data
+at the ``rx_data`` signal. The ``rx_valid`` signal stays asserted until the link
```

Comment:
I'm not sure I understand this phrase "to indicate that the peripheral is now providing and the processed data at the ``rx_data`` signal". Any thoughts on it? I think "and" doesn't make sense there